File: csharp/AntiMatter.cs
using System;

namespace NDP_Odev4
{
    internal class AntiMatter : QuantumObject, ICritical
    {
        public AntiMatter(string id, int stability, int dangerLevel) 
            : base(id, stability, dangerLevel) { }

        public override void Analyze()
        {
            int newValue = Stability - 25;
            if (newValue <= 0) throw new QuantumCollapseException(ObjectID);

            Stability = newValue;
            Console.WriteLine("The fabric of the universe trembles during analysis...");
        }

        public void EmergencyCooling()
        {
            int newValue = Stability + 50;
            // Safety Check: Prevent cooling if it results in an overflow beyond limits
            if (newValue >= 100)
            {
                Console.WriteLine("Warning: Cooling limit reached. Further cooling impossible.");
                return;
            }
            Stability = newValue;
            Console.WriteLine($"Thermal extraction complete. Stability: {Stability}%");
        }
    }
}

File: java/AntiMatter.java
// AntiMatter.java
public class AntiMatter extends QuantumObject implements ICritical {
    public AntiMatter(String id, int stability, int dangerLevel) {
        super(id, stability, dangerLevel);
    }

    @Override
    public void analyze() throws QuantumCollapseException {
        int newValue = getStability() - 25;
        if (newValue <= 0) throw new QuantumCollapseException(getObjectID());
        setStability(newValue);
        System.out.println("The fabric of the universe trembles during analysis...");
    }

    @Override
    public void emergencyCooling() {
        int newValue = getStability() + 50;
        if (newValue >= 100) {
            System.out.println("Warning: Cooling limit reached. Further cooling impossible.");
        } else {
            setStability(newValue);
            System.out.println("Thermal extraction complete. Stability: " + getStability() + "%");
        }
    }
}